Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals.h

```diff
@@ -232,7 +234,19 @@ namespace AzToolsFramework
         // ---------------- INTERNAL -----------------------------
         virtual bool HandlesType(const AZ::Uuid& id) const override
         {
-            return GetHandledType() == id;
+            if (auto applicationInterface = AZ::Interface<AZ::ComponentApplicationRequests>::Get(); applicationInterface != nullptr)
+            {
+                if (AZ::SerializeContext* serializeContext = applicationInterface->GetSerializeContext(); serializeContext != nullptr)
+                {
+                    // Look up the reflected SerializeContext class data in order to find the RttiHelper
+                    const AZ::SerializeContext::ClassData* classData = serializeContext->FindClassData(id);
+                    const AZ::IRttiHelper* rttiHelper = classData != nullptr ? classData->m_azRtti : nullptr;
+                    return id == AZ::GetAssetClassId() ||
+                        (rttiHelper != nullptr && rttiHelper->GetGenericTypeId() == AZ::GetAssetClassId());
+                }
+            }
+
+            return false;
```

Comment:
This still should check that the `AZ::SerializeTypeInfo<PropertyType>::GetUuid() == id`